scenes/music_puzzle/musician.dialogue

```diff
@@ -1,16 +1,39 @@
 ~ start
+if puzzle.is_solved():
+	=> well_done
+
+if not first_conversation:
+	=> hello_again
+
+set first_conversation = false
+
 Musician: Hello there, Storyweaver.
-Musician: Have you seen those stones all over this island?
-- What about them? They're just inert lumps of rock.
-	Musician: Oh, I wouldn't be so sure about that!
-- There's something unusual about them… it's like they have a certain energy.
-Musician: They're tuned to a major scale!
-	Legend has it that playing the right melody will open a door.
+if puzzle.get_progress() >= 1:
+	Musician: Oh! Have you been playing with the stones over there?
+	Player: I kicked them and something strange happened...
```

Comment:
I don't think it's necessary here but you could imagine the player having a name, and it turns out that you can use `{{ variable access }}` in the "character name" position, i.e. before the colon.

https://github.com/nathanhoad/godot_dialogue_manager/blob/main/docs/Basic_Dialogue.md#variables-in-dialogue

So the way to do this (in future) would be to add the player who is doing the interaction that triggered the dialogue to the extra_game_states array too.